src/main/kotlin/org/rust/lang/utils/RsDiagnostic.kt

```diff
@@ -99,10 +99,17 @@ sealed class RsDiagnostic(
         }
 
         private fun expectedFound(expectedTy: Ty, actualTy: Ty): String {
-            val expectedTyS = escapeString(expectedTy.toString())
-            val actualTyS = escapeString(actualTy.toString())
+            val expectedTyS = escapeTy(expectedTy.toString())
+            val actualTyS = escapeTy(actualTy.toString())
             return "expected `$expectedTyS`, found `$actualTyS`"
         }
+
+        // BACKCOMPAT: ???
+        // Fix for IntelliJ platform bug: https://youtrack.jetbrains.com/issue/IDEA-186991
+        // replace it with `escapeString()` after the end of support IDEs with the bug
+        private fun escapeTy(str: String): String = str
+            .replace("<", "&#10094;")
+            .replace(">", "&#10095;")
```

Comment:
You can use code of common angle brackets (60 and 62) here - it works fine with such escaping

Reply:
And are you sure that it is enough for escaping? what about `&`?
maybe create a copy of `XmlStringUtil.escapeString` with correct escaping of `<` and `>`?